distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Vector.enso

```diff
@@ -393,12 +421,42 @@ type Vector
 
              [1, 2, 3].to_text == "[1, 2, 3]"
     to_text : Text
-    to_text =
-        if this.length == 0 then "[]" else
-            if this.length == 1 then "[" + (this.at 0 . to_text) + "]" else
-                folder = str -> ix -> str + ", " + (this.at ix).to_text
-                tail_elems = 1.up_to this.length . fold "" folder
-                "[" + (this.at 0 . to_text) + tail_elems + "]"
+    to_text = this.generic_to_text "[" ", " "]"
+
+    ## PRIVATE
+       A generic utility which concatenates textual representations of the
+       vector elements, separated by the `separator` and appends the provided
+       `prefix` and `suffix`.
+    generic_to_text : Text -> Text -> Text -> Text
+    generic_to_text prefix separator suffix =
+        if this.length == 0 then prefix+suffix else
+            folder = str -> ix -> str + separator + (this.unsafe_at ix).to_text
+            tail_elems = 1.up_to this.length . fold "" folder
+            prefix + (this.unsafe_at 0 . to_text) + tail_elems + suffix
+
+    ## UNSTABLE
+       Generates a human-readable text representation of the vector, keeping its
+       length limited.
+
+       Arguments:
+       - max_entries: The maximum number of entries that are displayed. If the
+         vector contains more elements, the number of hidden elements is also
+         displayed.
+
+       > Example
+         Convert a large vector of numbers to a short text.
+
+             (0.up_to 100).to_vector.short_display_text max_entries=2 == "[0, 1 and 98 more elements]"
+    short_display_text : Integer -> Text
+    short_display_text max_entries=10 =
+        # TODO should this be some specific error type? We may want a generic Illegal_Argument_Error for such situations
```

Comment:
Todo left in the code. This should be fixed before merge.

Reply:
Thanks for catching it.
`Illegal_Argument_Error` type has been added.